ci/L0_multi_gpu/vllm_backend/vllm_multi_gpu_test.py

```diff
@@ -118,6 +123,59 @@ def _test_vllm_model(self, send_parameters_as_tensor=True):
 
         self.triton_client.stop_stream()
 
+    def test_multi_gpu_model(self):
+        """
+        Tests that a multi-GPU vLLM model loads successfully on multiple GPUs
+        and can handle a few sanity check inference requests.
+
+        Multi-GPU models are currently defined here as either:
+          - a single model instance with tensor parallelism > 1
+          - multiple model instances each with tensor parallelism == 1
+
+        FIXME: This test currently skips over a few combinations that may
+        be enhanced in the future, such as:
+          - tensor parallel models with multiple model instances
+          - KIND_MODEL models with multiple model instances
+        """
+        model = os.environ.get("TEST_MODEL")
+        kind = os.environ.get("KIND")
+        tp = os.environ.get("TENSOR_PARALLELISM")
+        instance_count = os.environ.get("INSTANCE_COUNT")
+        for env_var in [model, kind, tp, instance_count]:
+            self.assertIsNotNone(env_var)
+
+        print(f"Test Matrix: {model=}, {kind=}, {tp=}, {instance_count=}")
+
+        # Only support tensor parallelism or multiple instances for now, but not both.
+        # Support for multi-instance tensor parallel models may require more
+        # special handling in the backend to better handle device assignment.
+        # NOTE: This eliminates the 1*1=1 and 2*2=4 test cases.
+        if int(tp) * int(instance_count) != 2:
+            msg = "TENSOR_PARALLELISM and INSTANCE_COUNT must have a product of 2 for this 2-GPU test"
+            print("Skipping Test:", msg)
+            self.skipTest(msg)
```

Comment:
Can we put this into `@unittest.skipIf` ? It would be easier to locate then

Reply:
I think I'd have to move the tp and instance_counts to be global or passed directly to the test somehow to do this - I was trying to avoid being too fancy with these tests, but looks like I'll need to rethink them based on the comments so far.